testing/execution.ts

```diff
@@ -951,3 +962,168 @@ class DeveloperError extends Error {
+          formulaSpecification,
+          params,
+          bundleSourceMapPath,
+          bundlePath,
+          executionContext,
+        })
+      : await executeFormulaOrSyncWithRawParams({formulaSpecification, params, manifest, executionContext});
+
+    result.push(...response.result);
+    if (chainedCommand) {
+      chainedCommandResults.push(
+        ...(await chainCommandOnSyncResult({
+          rows: response.result,
+          formulaSpecification,
+          chainedCommand,
+          params,
+          manifest,
+          executionContext,
+          vm,
+          bundleSourceMapPath,
+          bundlePath,
+        })),
+      );
+    }
+    executionContext.sync.continuation = response.continuation;
+    iterations++;
+  } while (executionContext.sync.continuation && result.length < maxRows);
+  if (result.length > maxRows) {
+    result = result.slice(0, maxRows);
+  }
+
+  return {result, chainedCommandResults};
+}
+
+/**
+ * This function handles running a chained command formula for a given set of rows.
+ *
+ * @param rows The rows to run the chained command on
+ * @param formulaSpecification The Sync formula specification that is driving the rows we want to run
+ *    the chained command on.
+ * @param chainedCommand The chained command to run after the formula specification
+ * @param params The params to pass to the formula
+ * @param manifest The manifest of the pack
+ * @param executionContext The execution context
+ * @param vm Whether to run in a VM
+ * @param bundleSourceMapPath The source map path
+ * @param bundlePath The bundle path
+ * @returns Returns the result from the chained command
+ */
+async function chainCommandOnSyncResult({
+  rows,
+  formulaSpecification,
+  chainedCommand,
+  params,
+  manifest,
+  executionContext,
+  vm,
+  bundleSourceMapPath,
+  bundlePath,
+}: {
+  rows: any[];
+  formulaSpecification: SyncFormulaSpecification;
+  chainedCommand: FormulaSpecification;
+  params: string[];
+  manifest: BasicPackDefinition;
+  executionContext: SyncExecutionContext;
+  vm?: boolean;
+  bundleSourceMapPath: string;
+  bundlePath: string;
+}) {
+  if (!chainedCommand) {
+    return [];
+  }
+
+  // Denormalize the sync result before passing back into chained command
+  const formula = findSyncFormula(manifest, formulaSpecification.formulaName, executionContext.authenticationName);
+  const schema = executionContext.sync.schema ?? formula?.schema;
+  const denormalizedSyncResult = vm ? rows : untransformBody(rows, schema);
+
+  switch (chainedCommand.type) {
+    case FormulaType.GetPermissions:
+      const mappedRows = denormalizedSyncResult.map((row: unknown) => ({row}));
+
+      // 10 hardcoded as fallback to match process_csb_ingestion.ts
+      const maxPermissionBatchSize = formula?.maxPermissionBatchSize || 10;
+      const chunks = chunkArray(mappedRows, maxPermissionBatchSize);
+
+      const chunkResponses = await Promise.all(
```

Comment:
I would just do a `for (const chunk of chunks) {`, no need to do the chunks in parallel, the production doe does the chunks serially and I think it'll be easier to follow/debug as well.